ui/components/app/custom-spending-cap/custom-spending-cap.js

```diff
@@ -146,25 +159,24 @@ export default function CustomSpendingCap({
         >
           <label
             htmlFor={
-              value > (currentTokenBalance || error)
+              Number(value) > (currentTokenBalance || error)
                 ? 'custom-spending-cap-input-value'
                 : 'custom-spending-cap'
             }
           >
             <FormField
-              numeric
               dataTestId="custom-spending-cap-input"
               autoFocus
               wrappingLabelProps={{ as: 'div' }}
               id={
-                value > (currentTokenBalance || error)
+                Number(value) > (currentTokenBalance || error)
```

Comment:
I have a doubt, will `Number(...)` be able to take care of very big numbers ?

Reply:
Hi @jpuri, See the current situation in the video. This is an enormous number... Do you have any suggestions or should it stay as it is currently on develop?

https://user-images.githubusercontent.com/63151811/206159552-1c0953eb-338b-4e52-89f1-ed25d7b0fadc.mov

Reply:
The suggestion would be that we should use our number utilities, first off can you tell me what *type* of value 'currentTokenBalance' is? Is it a plain javascript number? Typically we have these as BigNumbers or BNs or even strings that need to be converted to BigNumbers/BN so that they can be compared/manipulated. 

Currently if  a user happens to have more than `9007199254740991` of a token the conversion to a number will break down the accuracy of the number and the comparison against their  total will be incorrect. That is what @jpuri is saying when she asks if it can handle very big numbers.

Number('10000000000000000000000000000000') will return a number but the accuracy of the conversion is not guaranteed and comparisons with that number and another number are also not guaranteed. MAX_SAFE_INTEGER is a constant specified by the language constraints of JavaScript. Anything over that target `9007199254740991` (currently) will not guarantee accurate results. So the advice here would be to convert the strings using `conversionGreaterThan` from `shared/modules/conversion.utils.js` which has examples throughout the codebase on usage, here is one:
```
  return conversionGreaterThan(
    { value: minimumTxCostInHexWei, fromNumericBase: 'hex' },
    { value: ethBalance, fromNumericBase: 'hex' },
  );
```

in this we are converting two numbers that are hexes to BigNumber types so that they can be accurately compared regardless of how high the value goes.

Reply:
@VSaric @adnansahovic I agree with @brad-decker here

It would be good if we could replace all cases of ` > ` comparisons with our helper function that uses bignumber internally `conversionGreaterThan`

To simplify things in these files, you could create a wrapper around the `conversionGreaterThan` function that sets `fromNumericBase` to `'dec'` for both numbers, so that you can just call something like `decConversionGreaterThan(tokenValue, currentTokenBalance)`

Or that might be unecessary if you do the comparison once and store the result in a variable and then use that variable throughout the file (because it seems we are doing the same comparison throughout the file)